modern/src/map/core/useMapStyles.js

```diff
@@ -169,20 +171,27 @@ export default () => {
       available: !!mapboxAccessToken,
       attribute: 'mapboxAccessToken',
     },
-    {
-      id: 'mapboxSatelliteStreet',
+    mapboxSatellite: {
+      id: 'mapboxSatellite',
       title: t('mapMapboxSatellite'),
       style: styleCustom(
         [`https://api.mapbox.com/styles/v1/mapbox/satellite-streets-v11/tiles/{z}/{x}/{y}?access_token=${mapboxAccessToken}`],
       ),
       available: !!mapboxAccessToken,
       attribute: 'mapboxAccessToken',
     },
-    {
+    custom: {
       id: 'custom',
       title: t('mapCustom'),
       style: styleCustom([customMapUrl]),
       available: !!customMapUrl,
     },
-  ];
+  };
+
+  Object.keys(Styles).forEach((key) => {
```

Comment:
I think the right think is to keep the array as is and just filter it. Otherwise the order is not guaranteed and you kind of have duplication of the ids.

Reply:
Are you suggesting that we use an array for `Styles` (like it currently is in master) and delete styles from it if found in the `disabledStyles` list? Or do you suggest that we still loop over `Styles` and create `activeStyles`, but only changing `Styles` back to a list instead of an dict?

I thought about looping though the `disabledStyles` set, looking for the id in `Styles` and removing it if found. The problem is that it would require a search though the entire `Styles` array for every entry in `disabledStyles`. Since `disabledStyles` will always be smaller than `Styles`, I decided that it would be better to perform the "has key" check on the `disabledStyles` set.

Changing `Styles` back to an array and still using the same logic where its copied over to `activeStyles` if not in  `disabledStyles`, should solve your order and id duplication concerns.

Reply:
> Changing `Styles` back to an array and still using the same logic where its copied over to `activeStyles` if not in `disabledStyles`, should solve your order and id duplication concerns.

I made this change with [d17e1b3](https://github.com/traccar/traccar-web/pull/970/commits/d17e1b3e2d6d539695e03c49c6c30b1481c06750)